client/signup/steps/domains/index.jsx

```diff
@@ -1349,6 +1350,11 @@ export class RenderDomainsStep extends Component {
 		} else if ( isOnboardingGuidedFlow( flowName ) ) {
 			// Let the framework decide the back url.
 			backUrl = undefined;
+		} else if ( isOnboardingFlow( flowName ) ) {
+			const searchParams = new URLSearchParams( window.location.search );
+			const existingParams = Object.fromEntries( searchParams.entries() );
+			backUrl = getStepUrl( flowName, 'designSetup', null, this.getLocale(), existingParams );
```

Comment:
Was unsure how we would call the getBack navigation from the stepper framework here, so hardcoding the previous step cc: @Automattic/vertex

Reply:
Passed the `goBack` function by [7b8c0e9](https://github.com/Automattic/wp-calypso/pull/97592/commits/7b8c0e9bca244714fd69d94c54b0edf3b2acd131).